scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -735,7 +737,7 @@ object FormatWriter {
   }
 
   private val leadingAsteriskSpace =
-    Pattern.compile("(?<=\\n)\\h*(?=[*][^*])")
```

Comment:
Can you explain why allowing match on multiple asterisks is equivalent? I tried yesterday, there was a test which failed but looks like for you it didn't.

Reply:
I took the requirement that " *X" gets replaced and " **" (double asterisk) does not as
"Do not even _think_ about breaking this" lore.  The match in the capture group
should match anything but what would be a second asterisk [^*].

Reply:
ah, was looking at the code on my phone, didn't see the `[^*]`. never mind.

`\n` vs `\\n`, any difference?

Reply:
> `\n` vs `\\n`, any difference?

thanks for adding tests. this question above is the last one.

Reply:
re: `\n vs \\n, any difference?`   

Pardon me for being concrete, no disrespect intended.  The concrete answer
is more of my inability to organize my thoughts above that. I have spent 
hours, days, or, more probably weeks, being hosed by exactly this issue.
The recurrence interval between episodes of my being hosed is just slightly
longer than my memory of the solution.   Mea culpa!

It is my understanding that for 'normal' strings  `\n` is always used and gets translated
to one standard Java two-octet Character.

In s-interpolated strings (and possibly/probably other interpolations) things get
a bit more confusing, at least for me.  I believe that a `\n` gets translated as for
a simple string. In `\\n` the first backslash quotes the second backslash, which
then becomes `\n`  which Pattern.compile interprets as though the original
string had been `\n`.

So, both work in interpolated strings, unless you have evidence or experience to
the contrary.  I tend to prefer the single `\n`  because it has fewer characters and transformations to parse.

Reply:
> re: `\n vs \\n, any difference?`
> 
> So, both work in interpolated strings, unless you have evidence or experience to
> the contrary. I tend to prefer the single `\n` because it has fewer characters and transformations to parse.

like `\\s`, `\\h` or other special escape sequences, i was curious whether `\\n` is interpreted by `Pattern` as anything different than simple `0xA`. clearly, `\n` is directly substituted as 0xA so pattern only sees that.

Reply:
Re: tests 

I found StripMargin.stat and added tests for only the `align` aspect. I did not address
the `no align` case, which is a weasel, driven by limited time.

I had expected to find tests for leading and trailing spaces. I could not find anything relevant
after a number of trials: using find for filenames and/or grepping for "whitespace" and
combinations.

I did find a Unicode.stat and could slip in a case for leading and trailing horizontal
whitespace containing Unicode there.  It would seem more integrated to have
it in a leadingWhitespace.spec if such a thing exists.  Would, IMHO, make it easier
for future visiting devos, such as myself, to find.

Off the top of your head, do you have any clues or suggestions.

Reply:
> Off the top of your head, do you have any clues or suggestions.

I added one tab in the `default/Comment` test, see at the bottom:

https://github.com/scalameta/scalafmt/pull/1911/files

you can create a dedicated whitespace test but i considered it part of handling comments and strings.

Reply:
@kitbellew The info you gave helped me to get to the next round.  Thank you.

All tests, new & previous,  pass on my system.  I will check CI later.

The tests are still sensitive to some future devo removing the now invisible whitespace
especially the tabs.

I think the three regex expressions are as simple & fast as the requirements and reasonable
end use allows.  I left a comment in FormatWriter.scala explaining why. Figured that might
endure long after the present conversation is forgotten.

I suspect that there are a couple of bugs remaining in scalafmt relating to tabs. I will see in
my mainline daily work  if I can trick them from coming out of the dim shadows.

Unit tests prove their worth, much as they are a pain to write.  The hard time I had
getting unit tests to accept Ogham marks and unicode \uMumble convinced me
that a simple "space or tab" test is all that is needed.  There is nothing wrong with
"\h", just that it will spend a lot of time testing against characters which can not be 
in the input.   Sorry if I lead you down a loosing path.

If you like the current work, I can update & buff the commit message and change the
title.